qa/restricted-loggers/src/test/java/org/elasticsearch/common/logging/LoggersTests.java

```diff
@@ -69,36 +69,48 @@ public void testSetLevelWithRestrictions() {
         assertThat(Loggers.RESTRICTED_LOGGERS, hasSize(greaterThan(0)));
 
         for (String restricted : Loggers.RESTRICTED_LOGGERS) {
+            Level rootLevel = LogManager.getRootLogger().getLevel();
 
             // 'org.apache.http' is an example of a restricted logger,
             // a restricted component logger would be `org.apache.http.client.HttpClient` for instance,
             // and the parent logger is `org.apache`.
             Logger restrictedLogger = LogManager.getLogger(restricted);
+            Level restrictedLoggerLevel = restrictedLogger.getLevel();
```

Comment:
Does getLevel return null if the effective level is inherited? Otherwise it seems we might force the current parent log level to be realized on this logger when restoring it.

Reply:
Good point, it's always the effective log level. 
Also, when setting root to DEBUG, all the restricted loggers will be configured with level INFO. So there's a lot more to do to revert any possible changes. I've added a small util `TestLoggers.runWithLoggersRestored` to do that.